sktime/forecasting/varmax.py

```diff
@@ -0,0 +1,392 @@
+        signal_only=False,
+    ):
+        # Model parameters
+        self.order = order
+        self.trend = trend
+        self.error_cov_type = error_cov_type
+        self.measurement_error = measurement_error
+        self.enforce_stationarity = enforce_stationarity
+        self.enforce_invertibility = enforce_invertibility
+        self.trend_offset = trend_offset
+        self.start_params = start_params
+        self.transformed = transformed
+        self.includes_fixed = includes_fixed
+        self.cov_type = cov_type
+        self.cov_kwds = cov_kwds
+        self.method = method
+        self.maxiter = maxiter
+        self.full_output = full_output
+        self.disp = disp
+        self.callback = callback
+        self.return_params = return_params
+        self.optim_score = optim_score
+        self.optim_complex_step = optim_complex_step
+        self.optim_hessian = optim_hessian
+        self.flags = flags
+        self.low_memory = low_memory
+        self.dynamic = dynamic
+        self.information_set = information_set
+        self.signal_only = signal_only
+
+        super(VARMAX, self).__init__()
+
+    def _fit_forecaster(self, y, X=None):
+        """Fit forecaster to training data.
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        y : array_like
+            The observed time-series process :math:`y`, shaped n_obs x k_endog.
+        X : array_like, optional (default=None)
+            Array of exogenous regressors, shaped n_obs x k.
+
+        Returns
+        -------
+        self : reference to self
+        """
+        self._forecaster = _VARMAX(
+            endog=y,
+            exog=X,
+            order=self.order,
+            trend=self.trend,
+            error_cov_type=self.error_cov_type,
+            measurement_error=self.measurement_error,
+            enforce_stationarity=self.enforce_stationarity,
+            enforce_invertibility=self.enforce_invertibility,
+            trend_offset=self.trend_offset,
+        )
+        self._fitted_forecaster = self._forecaster.fit(
+            start_params=self.start_params,
+            transformed=self.transformed,
+            includes_fixed=self.includes_fixed,
+            cov_type=self.cov_type,
+            cov_kwds=self.cov_kwds,
+            method=self.method,
+            maxiter=self.maxiter,
+            full_output=self.full_output,
+            disp=self.disp,
+            callback=self.callback,
+            return_params=self.return_params,
+            optim_score=self.optim_score,
+            optim_complex_step=self.optim_complex_step,
+            optim_hessian=self.optim_hessian,
+            flags=self.flags,
+            low_memory=self.low_memory,
+        )
+        return self
+
+    # defining _predict method to pass in dynamic, information_set and signal_only
+    # which are not passed into the _predict method in _StatsModelsAdapter
```

Comment:
suggestion: adjust this as there are now *two* reasons why we need this. 
```suggestion
    # defining `_predict`, instead of inheriting from `_StatsModelsAdapter`,
    # for two reasons:
    # 1. to pass in `dynamic`, `information_set` and `signal_only`
    # 2. to deal with statsmodel integer indexing issue
```